### PR TITLE
Support for multiple private registries in 2.6 (secretmigrator code)

### LINKED ISSUE #37658: [2.6] Support for multiple private registries is broken
**Rancher Server Setup**
- Rancher version: 2.6.4
   
**Describe the bug**

RKE supports multiple private registries:
https://rancher.com/docs/rke/latest/en/config-options/private-registries/

1. Create or edit a cluster/rke cluster template and directly modify the RKE yaml.
2. Click Edit Yaml button and create multiple registry entries like so

```
  private_registries:
    - ecr_credential_plugin: {}
      is_default: true
      url: my-private-registry
      user: tobias
      password: secure
    - ecr_credential_plugin: {}
      is_default: false
      url: my-private-registry-2
      user: frank
      password: securepw2
```
3. Try to submit it and you will be greeted with an error "resourceVersion should not be set on objects to be created"

Relevant broken function:
https://github.com/rancher/rancher/blob/dfe44512d12d5674607be1848e53ce83f5b67e6c/pkg/controllers/management/secretmigrator/clusters.go#L661

Should be assigned to @cmurphy as she wrote the secret migrator code.

### PR DESCRIPTION
**Pull request by zerotens**
#37658 

It will fix the logic of the function `CreateOrUpdatePrivateRegistrySecret` to store secrets for multiple registries.
https://github.com/rancher/rancher/blob/dfe44512d12d5674607be1848e53ce83f5b67e6c/pkg/controllers/management/secretmigrator/clusters.go#L661

The relevant secrets in the namespace `cattle-global-data` will be created correctly.

### CODE REVIEW

File: pkg/controllers/management/secretmigrator/clusters.go [hunk 1]
```diff
@@ -702,60 +703,48 @@ func (m *Migrator) CreateOrUpdatePrivateRegistrySecret(secretName string, rkeCon
...
 			continue
 		}
-		registry := credentialprovider.DockerConfigJSON{
-			Auths: credentialprovider.DockerConfig{
-				privateRegistry.URL: credentialprovider.DockerConfigEntry{
-					Username: privateRegistry.User,
-					Password: privateRegistry.Password,
-				},
-			},
+		// limitation: if a URL is repeated in the privateRegistries list, it will be overwritten in the registry secret)
```
Comment (cmurphy) [inline pkg/controllers/management/secretmigrator/clusters.go:717]: (nit) there's an extra parenthesis that wasn't in the original comment

**Review by cmurphy — COMMENTED** [review]
This looks great, thanks for catching this. Could you squash the commits into a single commit?

**Review by MbolotSuse — COMMENTED** [review]
I have no issues outside of what @cmurphy has already noted (squash commits, remove paren in comment).

Feel free to re-request once you have resolved those issues. Thank you for taking the time to submit a pr for this issue.

File: pkg/controllers/management/secretmigrator/clusters.go (hunk 1 above)
Comment (zerotens) [reply pkg/controllers/management/secretmigrator/clusters.go]: Thanks for catching this. Removed the extra parenthesis and squashed the commits into one.

**Review by cmurphy — APPROVED** [review]

**Review by cmurphy — CHANGES_REQUESTED** [review]
We can merge this now, but it looks like you ran a git pull on this branch and added a merge commit to the PR. Can you remove the merge commit with

```
git fetch origin
git rebase origin/release/v2.6
```

Thanks!

**Comment by zerotens** [comment]
> We can merge this now, but it looks like you ran a git pull on this branch and added a merge commit to the PR. Can you remove the merge commit with
> 
> ```
> git fetch origin
> git rebase origin/release/v2.6
> ```
> 
> Thanks!

@cmurphy 

My bad, should be fixed now.

**Review by cmurphy — APPROVED** [review]

**Pull request merged**